content/blog/nuttx-using-wokwi/index.md

```diff
@@ -0,0 +1,222 @@
+
+  - Provides convenience in getting help and feedback through the link to the project.
+
+  - Helps distinguish hardware and software based problems. This way, you can focus on the code side.
+
+  - It allows the use of any number of hardware regardless of the project budget.
+
+<cite>[Source][3]</cite>
+
+Note: Of course, there will be differences between the simulation and the field. For this reason, the project running in simulation should not be taken to the field without testing.
+
+
+## Supported devices
+
+Here is the list of supported devices, separated by the core architecture.
+
+| SoC      | Core   | Support Stage |
+|----------|--------|---------------|
+| ESP32    | Xtensa | Release       |
+| ESP32-S2 | Xtensa | Release       |
+| ESP32-S3 | Xtensa | Release       |
+| ESP32-C3 | RISC-V | Release       |
+| ESP32-C6 | RISC-V | Release       |
+| ESP32-H2 | RISC-V | Beta          |
+| ESP32-P4 | RISC-V | Alpha         |
+
+
+<cite>[Source][4]</cite>. Apart from these, you can use [this link](https://docs.wokwi.com/getting-started/supported-hardware) for supported sensors and more.
+
+## Alternatives and Differences
+
+If we compare Wokwi with QEMU, which is not the only alternative but one of the most known one. **QEMU** stands out with faster operation with special optimizations while **Wokwi** offers more platform support, ease of installation and simulation features with external parts.
+
+## Simulating NuttX on Wokwi
+
+### Creating a NuttX System Image
+
+  - You can create the development environment for Xtensa-based (ESP32, ESP32-S2, ESP32-S3) microcontrollers using [this link](https://nuttx.apache.org/docs/latest/platforms/xtensa/index.html).
+  - You can create the development environment for RISC-V based (ESP32-C3, ESP32-C6, ESP32-H2) microcontrollers using [this link](https://nuttx.apache.org/docs/latest/platforms/risc-v/index.html).
+
+Wokwi does not support compiling NuttX projects internally like ESP-IDF, Zephyr. Therefore, it is necessary to compile locally and continue the simulation part via Wokwi. In other words, we will simulate the NuttX output compiled in our own development environment via Wokwi.
+
+In this article, we will use the nsh application for the esp32 microcontroller as an example. Enter the following commands to clean if there is a previously selected configuration and to select configuration needed for nsh application for esp32.
+
+```bash
+make distclean
+./tool/configure.sh esp32-devkitc:nsh
+```
+
+With these commands, we have selected the necessary card and configuration. To perform debugging, we need to activate the `Generate Debug Symbols` feature in the `Debug Options` section of the `Build System` menu. We can use the following command for the relevant setting:
+
+```bash
+make menuconfig
+```
+
+The result will look something like this:
+
+![make menuconfig](img/make_menuconfig.webp)
```

Comment:
Suggest that you make the window smaller and then take the screenshot, because the valuable information is the text in the middle which is very small.